hyp3lib/make_cogs.py

```diff
@@ -1,45 +1,41 @@
-"""Creates a Cloud Optimized Geotiff from the input geotiff(s)"""
+"""Creates a Cloud Optimized GeoTIFF from the input GeoTIFF(s)"""
 
 import argparse
-import glob
 import logging
 import os
 import shutil
+import sys
+from glob import glob
+from tempfile import NamedTemporaryFile
 
 from osgeo import gdal
 
 
-def cogify_dir(dir="PRODUCT",debug=False,res=30):
-    back = os.getcwd()
-    os.chdir(dir)
-    tmpfile = "tmp_cog_{}.tif".format(os.getpid())
-    for myfile in glob.glob("*.tif"):
-        logging.info("Converting file {} into COG".format(myfile))
-        make_cog(myfile,tmpfile,res=res)
-        shutil.move(tmpfile,myfile)
-    os.chdir(back)
+def cogify_dir(directory: str = "PRODUCT", file_pattern: str = '*.tif'):
```

Comment:
now that I think about it, `directory` really shouldn't have a default value.  I'm thinking of the poor developer encountering a line of code in the wild that's just `cogify_dir()`; no reason to expect that's cogifying the `PRODUCT` directory

Reply:
or maybe `directory='.'` or `directory=''` would be better, that would at least cog-ify the working directory

Reply:
my approval still stands, though, feel like we have have revised this PR to death already

Reply:
I agree with no default actually. I think as a library function you should have to be explicit